WordPress/openverse
# Add additional search view pages to the Nuxt app

~~This PR is based on #3136, and will be undrafted after that PR is merged.~~

## Fixes



Fixes #2858 by @obulat
Fixes #2774 by @obulat 

## Description



With this PR, you can view the tag/source/creator collections in the Nuxt app.

To show the collections, we add `_tag.vue` and `_source.vue` pages to both `image` and `audio`, with dynamic routes. These pages check the path to make sure that the path has the necessary parameters: tag name for the tag page; an existing source for source/creator pages; `/creator/` path part and the creator name for the creator pages. 

In the first iteration, I used a single page with a fully-dynamic path (`/_mediaType/_collection/_.vue`) that would then validate the `mediaType`, `collection` and the collection parameters. However, because `_mediaType` can accept any string path, it would also match the locale routes such as `/ar/non_existent_page`), so I split this single page into 4.

The collection pages have the collection headers (using `VCollectionHeader` from #3365) and either a `VAudioCollection` or `VImageGrid`, as well as the "Load more" button. When you navigate from the collection page to a single result page, you also get the "Back to results" link at the top that goes back to the collection page.

This PR adds a `strategy` and `collectionParams` props to the `search` store. Whenever we set the `searchTerm` (which is only available for search, not collection), we reset the `strategy` to `default` and nullify the `collectionParams`.

The API service had to be updated to accept a `slug` for paths like `/tag/tag_name`, `/source/source_name`, `/source/source_name/creator/creator_name`, which are added to the `/<mediaType>` path.
The media store was updated to use the necessary slug from the search store `collectionParams`.

## Testing Instructions

Run the app using `just frontend/run dev`. Go to `/preferences` page and turn on the `additional_search_views` feature flag.

The easiest way to land on a collection page is from the single result page. Click on the source, creator or tag buttons - and you will be taken to the corresponding page.
Check that in non-English locales, the links have the correct locale path, and when navigating, the locale is kept. 
Check that if you go from a collection page to a single result page, you can go back to the collection page using "Back to results" page.
Check that if you go back to homepage from the collection page, you can search for something.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
